### DISCUSSION TITLE
Invidividual Keys support for Fastlane match

### DISCUSSION THREAD

**Discussion by andrej-jasso** [fastlane/fastlane#22203] [Ideas]
Like it's mentioned [here](https://github.com/fastlane/fastlane/issues/21959#issuecomment-2220436870) the Match command still missing the individual key option as well.

Would be great if it were eventually added to the fastlane library.

Currently the API requires you to specify Issuer ID. but with individual keys there is no issuer ID as far as I can tell from the Appstore Connect web.

**Comment by bella964** [comment]
Hello @andrej-jasso 


You're correct, and this has been a notable limitation for some developers using Fastlane's match tool when dealing with App Store Connect's App Store Connect API.

The match tool currently requires the Issuer ID for creating and managing certificates, which aligns with team-based API keys. However, individual keys generated for a single app in App Store Connect don’t include an Issuer ID, as these keys are app-specific and not team-based.   [obamacare enrollment](https://www-obamacare.com)

Why is this important?
Team API Keys require an Issuer ID and can manage multiple apps across a team.
Individual App Keys do not use an Issuer ID and are specific to a single app.
If Fastlane were to support individual keys, it would:

Allow developers working on single-app projects to utilize match without team API keys.
Simplify workflows for apps managed independently.
Current Workaround
For now, developers using individual keys can:

Manually manage certificates for individual keys in App Store Connect.
Use fastlane commands like cert and sigh individually, bypassing match.
Feature Request Proposal
To request support for individual keys in the match command:

Open a feature request in Fastlane's GitHub issues page.
Provide details:
Explain the limitation with Issuer IDs and individual keys.
Share how individual key support could simplify workflows.
Suggest ways the Fastlane team could differentiate between Team API Keys and Individual Keys in the match command.


Best Regards
